docs/spec.md

```diff
@@ -446,18 +435,19 @@ to that of the derivation (or derivations).
 
 
 
-## Traversable
-
-#### Dependencies
-
-  1. Functor
-  1. Foldable
+### Traversable
 
-#### Methods
+In the following signature `Applicative<U>` means that a value must
+not only match `Applicative` signature, but also fully support `Applicative` algebra.
 
-  1. `traverse :: (Traversable t, Applicative f) => Type t ~> (Type f, (a → f b), t a) → f (t b)`
+```js
+Traversable<T> {
+  traverse: (Applicative<U>, a => U<b>, T<a>) => U<T<b>>
+}
+```
```

Comment:
This doesn't seem right. `U<*> extends Applicative<U>` suggest that `U` is a module, then `a => U<b>` and `U<T<b>>` are wrong, these functions are not returning modules. 

To clarify, in my current version `U` is a type like `Array`, e.g. if second argument of `traverse()` is a function that returns arrays, then first argument must be a module that supports `Applicative` with `T=Array`.